README.md

```diff
@@ -14,52 +14,164 @@ Backend developed by the Digital R&D team for hybrid search apps. Related repos:
 
 Install with `pip install git+ssh://git@github.com/ministryofjustice/rd-search-backend.git`
 
-## Dev
 
-Working on the app relies on Python dependencies being installed locally.
-First set up a virtualenv and install dependencies:
+## Quick start
+
+Steps to get started indexing and searching documents in local OpenSearch (see also notebooks/demo_search.ipynb):
+
+1. Open Docker Desktop
+
+2. Start the stack
 
 ```
-python3 -m venv venv
-source venv/bin/activate
-pip install -e '.[dev, search_backend]'
+docker compose up
 ```
 
-Prior to pushing commits to GitHub, run the pre-commit hooks with:
-`pre-commit run --all-files`
+3. Run imports and define config variables
 
-To build the package manifest with hatchling, run `hatchling build` from
-terminal at the project root. By default this writes to ./dist/, but can be
-directed with the `-d` flag.
+```
+from haystack import Document
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+from search_backend.indexing_pipeline import IndexingPipeline
+from search_backend.retrieval_pipeline import RetrievalPipeline
+from search_backend.search import Search
+
+
+cfg = {
+    # Optional arg for the OpenSearch docstore, to prevent trying to index everything in one go
+    "index_batch_size": 10,
+    # Select embedding model for the semantic search. This should be a sentence-similarity
+    # model available on Huggingface: https://huggingface.co/models?pipeline_tag=sentence-similarity
+    "dense_embedding_model": "sentence-transformers/all-MiniLM-L6-v2",
+    # The value of the embedding dimension must match that specified for the model defined above
+    "embedding_dim": 384,
```

Comment:
Interestingly, I tried varying this in the notebook expecting the model to complain that embeddings of different dimensions were being asked for, but it did not raise or affect the search outcomes.

Reply:
Hmm, I've tried using `embedding_dim` values of 100 and 500 with a model that wants 384 and it gave an error: `failed to parse field [embedding] of type [knn_vector] in document with id`. I needed to stop and recompose the docker image, otherwise it would try to use whatever previous `embedding_dim` was set.

Reply:
right, that's the behaviour I expected also. Was this in the same notebook here? I was going to suggest implementing some basic defences to check that the embeddings dimensions are appropriate to the model (or even derive them from querying the model metadata if possible), but then found it curious that adjusting the config dictionary didn't raise an issue.

Reply:
Yeah that's using the same notebook. Yeah it'd be ideal if we could derive the embedding dimensions - I haven't seen it done with Haystack but will have a scout around

Reply:
Looks like in theory you should be able to get model metadata from the HuggingFace API (https://huggingface.co/docs/huggingface_hub/v0.26.3/en/package_reference/hf_api#huggingface_hub.HfApi.list_datasets) but I haven't managed to get it to filter down to the model I want.

Here's what I've tried, which looks like it should work if I've understood the documentation, but doesn't return any results:
```
from huggingface_hub import HfApi 
api = HfApi()
models = api.list_models(filter=("model_name:all-MiniLM-L6-v2"))
list(models)
```

Reply:
That looks promising. Yeah if we could retrieve the model metadata we could remove the embeddings dimensions from the config, less to go wrong from the user perspective. Might be worth leaning into it.

Reply:
I'll resolve this conversation for now and spin up an issue